src/Datadog.Trace.ClrProfiler.Managed/ScopeFactory.cs

```diff
@@ -19,6 +19,56 @@ internal static class ScopeFactory
 
         private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.GetLogger(typeof(ScopeFactory));
 
+        public static Scope GetActiveHttpScope(Tracer tracer)
+        {
+            var scope = tracer.ActiveScope;
+
+            var parent = scope?.Span;
+
+            if (parent != null &&
+                parent.Type == SpanTypes.Http &&
+                parent.GetTag(Tags.InstrumentationName) != null)
+            {
+                return scope;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Creates a span context for outbound http requests, or get the active one.
+        /// Used to propagate headers without changing the active span.
+        /// </summary>
+        /// <param name="tracer">The tracer instance to use to create the span.</param>
+        /// <param name="integrationId">The id of the integration creating this scope.</param>
+        /// <returns>A span context to use to populate headers</returns>
+        public static SpanContext CreateHttpSpanContext(Tracer tracer, IntegrationInfo integrationId)
+        {
+            if (!tracer.Settings.IsIntegrationEnabled(integrationId))
+            {
+                // integration disabled, skip this trace
+                return null;
+            }
+
+            try
+            {
+                var activeScope = GetActiveHttpScope(tracer);
+
+                if (activeScope != null)
+                {
+                    return activeScope.Span.Context;
+                }
+
+                return tracer.CreateSpanContext(serviceName: $"{tracer.DefaultServiceName}-{ServiceName}");
```

Comment:
nit: You don't end up using the service name so you should be able to remove this without any issue

Reply:
Technically correct, but I'm afraid it could cause issues if somebody in the future decides to use `CreateHttpSpanContext` in a different context

Reply:
Yeah you're right, we should keep this behavior